include/bluetooth/radio_notification_cb.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BT_RADIO_NOTIFICATION_CB_H__
+#define BT_RADIO_NOTIFICATION_CB_H__
+
+/**
+ * @file
+ * @defgroup bt_radio_notification_cb Radio Notification callback
+ * @{
+ * @brief APIs to set up radio notification callbacks
+ *
+ * The radio notification callbacks are triggered relative to the
+ * start of a Link Layer event.
+ *
+ * This information can be used to reduce the time from data sampling
+ * until data is sent on air.
+ */
+
+#include <stdint.h>
+#include <zephyr/bluetooth/conn.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Recommended prepare distance for the connection event callback. */
+#define BT_RADIO_NOTIFICATION_CONN_CB_PREPARE_DISTANCE_US_RECOMMENDED 3000
+
+/** Radio Notification connection callbacks
+ *
+ * The callbacks will be called from the system workqueue.
+ */
+struct bt_radio_notification_conn_cb {
+	/** Radio Notification prepare callback
+	 *
+	 * The prepare callback will be triggered a configurable amount
+	 * of time before the connection event starts.
+	 *
+	 * @param[in] conn The connection context.
+	 */
+	void (*prepare)(struct bt_conn *conn);
+};
+
+/** Register a radio notification callback struct for connections.
+ *
+ * @param[in] cb                  The callback structure to be used.
+ *                                The memory pointed to needs to be kept alive by the user.
+ * @param[in] prepare_distance_us The distance in time from the start of the
+ *                                prepare callback to the start of the connection event.
+ *
+ *  @retval 0          The callback structure has been successfully registered.
+ *  @retval -EALREADY  A callback has already been registered.
+ *  @retval -ENOMEM    There are not enough PPI channels available to use this feature.
+ */
+int bt_radio_notification_conn_cb_register(struct bt_radio_notification_conn_cb *cb,
```

Comment:
```suggestion
int bt_radio_notification_conn_cb_register(struct bt_radio_notification_conn_cb const* cb,
```
so app can put it to flash

Reply:
I would change an order to `const struct`